### PR TITLE
Fix Email Input Type and Improve Error Messaging on Signup and Login Pages 

### LINKED ISSUE #129: Bug:  Email Input Type and Error Messaging on Signup and Login Pages , i can add beautiful toast message
Hello @algovengers,

 ## I am writing to report a couple of issues on the signup and login pages of the website:

### 1. Email Input Type Not Defined:
 The email input fields on both the signup and login pages are not defined as type "email," which can lead to invalid email formats being entered.
### 2. Error Messaging: 
After entering incorrect information, a small red text appears as an error message. While this indicates an error, it could be improved for a better user experience.

## I propose fixing these issues by:

1. Defining the email input fields as type "email" to ensure proper validation.

2. Replacing the small red text with a more user-friendly toast message for both error and successful login/signup events.

I am eager to work on this issue and provide a solution that enhances the user experience. Could you please assign this task to me? I am confident that I can deliver a high-quality fix.

## Screenshot -
To help illustrate the issue, I have attached a screenshot for your reference.

https://github.com/algovengers/Hairify/assets/145179532/e34bb5d9-022c-44a4-9ca0-9950f8e0ad2a

## Thank you for your attention to this matter. I look forward to your response.

### PR DESCRIPTION
**Pull request by Aniket-2504**
**Description:**
fixes #129 

This pull request addresses the issues described in issue #129 regarding the email input type and error messaging on the signup and login pages. 

### Changes Made:

1. **Email Input Type Defined:**
   - Updated the email input fields on both the signup and login pages to have the type "email". This ensures that the browser performs proper validation for email formats, preventing invalid emails from being entered.

2. **Improved Error Messaging:**
   - Replaced the small red text error messages with beautiful and user-friendly toast messages using `react-toastify`. This enhancement provides a better user experience by clearly indicating errors and successful actions in a more visually appealing way.

### Implementation Details:

- **Error Messaging:**
  - Integrated `react-toastify` to display toast messages for various scenarios:
    - Displaying error messages when required fields are empty.
    - Showing error messages for password mismatch.
    - Indicating successful signup with a toast message.
  - Included the `ToastContainer` component in the respective components to ensure that the toasts are rendered.

### Testing:

- Verified that invalid email formats are now rejected by the browser.
- Checked that toast messages are displayed for the following cases:
  - Empty email, username, or password fields.
  - Passwords do not match.
  - Successful signup or login.

### Screenshots:

Screenshot 2024-06-14 175026
Screenshot 2024-06-14 175005
Screenshot 2024-06-14 174948

### Conclusion:

These changes enhance the form validation and user feedback on the signup and login pages, providing a smoother and more intuitive user experience. I look forward to feedback and am confident that this fix will improve the overall usability of the application.

---

Please let me know if there are any additional changes or details needed.

### CODE REVIEW

File: frontend/src/app/signup/page.tsx [hunk 1]
```diff
@@ -29,21 +31,21 @@ function SignupInner() {
   const [Signingup, setSigningUp] = useState<boolean>(false);
 
   const handleSignup = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
-    e.preventDefault();
+    // e.preventDefault();
```
Comment (afeefuddin) [inline frontend/src/app/signup/page.tsx:34]: Any specific reason for this?

**Comment by Aniket-2504** [comment]
I removed e.preventDefault(); because it was handling the required error, which prevented the browser from showing it to the user. Now, the browser handles the required error now its shows properly  you can see it in screenshots,

**Comment by Aniket-2504** [comment]
Thank you bro for giving me this opportunity.

**Comment by Aniket-2504** [comment]
hello bro is everything all right ?

**Comment by afeefuddin** [comment]
The toast is looking pretty weird as per the design, can we just show the message there instead of showing the toast.

**Comment by Aniket-2504** [comment]
can i change the position of toast or show the message only ?

**Comment by afeefuddin** [comment]
Please show messages only!

**Comment by Aniket-2504** [comment]
ok bro thanks , ill work on it

**Pull request closed without merging**